doc/exts/pylint_extensions.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/master/COPYING
+
+"""Script used to generate the extensions file before building the actual documentation."""
+
+import os
+import sys
+
+import sphinx
+
+from pylint.lint import PyLinter
+
+
+def builder_inited(app):
+    # PACKAGE/docs/exts/pylint_extensions.py --> PACKAGE/
+    base_path = os.path.dirname(
+        os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+    # PACKAGE/ --> PACKAGE/pylint/extensions
+    ext_path = os.path.join(base_path, 'pylint', 'extensions')
+    modules = []
+    for filename in os.listdir(ext_path):
+        name, ext = os.path.splitext(filename)
+        if ext != '.py' or name[0] == '_':
+            continue
+        # skip renamed deprecated modules to avoid conflicts with new modules
+        if name == "check_docs":
+            continue
+        modules.append('pylint.extensions.%s' % name)
+    if not modules:
+        sys.exit("No Pylint extensions found?")
```

Comment:
I was thinking that this should be treated as an error - we should always have at least one Pylint extension, so if we don't find any at all, something's gone wrong. I can change it to a return if you think this is something we should treat as valid instead.